### ISSUE TITLE
Unread messages aren't marked as read on opening

### ISSUE THREAD

**Issue by astatine** [FossifyOrg/Messages#264]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

ArrowOS 13.1 (Android 13 Custom ROM)

### Affected device model

Xiaomi Mi A2

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

Unread messages aren't marked as read on opening, i.e., they are still **bold** in the message list.

Check the attached screen recording. Sorry about the quality, I wanted to blur stuff out and didn't know what I was doing.

This is not the only 'not-actually-unread' message in my list, there are other messages which are also afflicted by this malaise.

### Expected behavior

Unread messages get marked as read on opening.

### Actual behavior

Unread messages aren't marked as read on opening

### Additional information

https://github.com/user-attachments/assets/fb5db53b-e150-496c-ad1b-36713f20a284

**Comment by naveensingh** [comment]
There is probably at least one message in that thread that was never "read". The app should provide better hints but the messages are marked when they are actually fetched from the database.

Workaround: Try scrolling up for a couple pages and the unread message should get marked as read automatically.

**Comment by astatine** [comment]
@naveensingh Thanks for the reply. I scrolled all the way up to the first message in that conversation, and scrolled back down. I did this multiple times. It didn't work.

I then scrolled down my message list and opened a 'not-actually-unread' conversation with only one message. It didn't work here either; going back to the conversation list still showed that conversation in bold.

For now, I've selected all conversations and force-marked them as read to mollify my offended sensibilities. ;-)

**Comment by naveensingh** [comment]
> I then scrolled down my message list and opened a 'not-actually-unread' conversation with only one message. It didn't work here either; going back to the conversation list still showed that conversation in bold.

must be something else then :)

**Comment by astatine** [comment]
@naveensingh @Aga-C It is happening again with v1.1.0 as well. Anything I can do to help you debug?

**Comment by naveensingh** [comment]
Thanks but it hasn't been fixed yet. We'll ping you if we need more help.

**Comment by astatine** [comment]
> Thanks but it hasn't been fixed yet. We'll ping you if we need more help.

So you're able to reproduce it? Just curious. Thanks!

**Comment by Aga-C** [comment]
@astatine I was able to reproduce it, that's why I've taken off `needs triage` label. However, I haven't encountered it for the one-message conversation as you described later.

**Comment by naveensingh** [comment]
> I haven't encountered it for the one-message conversation as you described later.

Same, the workaround works for me.

Would it be better if have something like an overlay with "5 unread messages ↑" or should the app mark everything as read automatically even if the user hasn't scrolled to actually "read" the message?

**Comment by tswistak** [comment]
> Would it be better if have something like an overlay with "5 unread messages ↑" or should the app mark everything as read automatically even if the user hasn't scrolled to actually "read" the message?

How about both at once 😄? When I enter the conversation, it shows where are unread messages (they could be marked with some different background too), but leaving the conversation marks all as read.

**Comment by naveensingh** [comment]
> How about both at once 😄? When I enter the conversation, it shows where are unread messages ... but leaving the conversation marks all as read

Yeah, this seems like an acceptable way to implement it without it being 'annoying'.

> it could scroll to the oldest unread message

I doubt everyone will like that, we'll let the overlay button with the arrow do the scrolling.

> different background

Yep, a slight hint is a good idea. We can fade the unread message from, let's say, 80% opacity to 100%.

**Comment by astatine** [comment]
> However, I haven't encountered it for the one-message conversation as you described later.

@naveensingh @Aga-C I caught a live one! Hope this helps.

https://github.com/user-attachments/assets/fd4a6ff3-8697-4e88-a0c2-565824e75d40

**Comment by naveensingh** [comment]
It is all pixelated but I can see it, thanks.

**Comment by astatine** [comment]
> It is all pixelated but I can see it, thanks.
It is pixelated on purpose for privacy reasons, as before.

I just noticed that a message _previously forcefully marked as read_ is now showing as unread. As before, it does not get marked as read on opening it. Maybe this is related to the other zombie ~~undead~~ unread messages?

**Comment by naveensingh** [comment]
@astatine just got one more question, how many conversations do you have?

**Comment by jrockoio** [comment]
I have maybe 200-300 conversations and get the bug frequently

**Comment by naveensingh** [comment]
Similar new issue: https://github.com/FossifyOrg/Messages/issues/609
